File: Lab 2/Owais Quadri/Node/producer.js
//const Kafka = require("kafkajs").Kafka
const {Kafka} = require("kafkajs")
const msg = process.argv[2];
run();
async function run(){
    try
    {
         const kafka = new Kafka({
              "clientId": "myapp",
              "brokers" :["localhost:9093","localhost:9094","localhost:9095"]
         })

        const producer = kafka.producer();
        console.log("Connecting.....")
        await producer.connect()
        console.log("Connected!")
        //A-M 0 , N-Z 1 
        const partition = msg[0] < "N" ? 0 : 1;
        const result =  await producer.send({
            "topic": "Test",
            "messages": [
                {
                    "value": msg,
                    "partition": partition
                }
            ]
        })

        console.log(`Send Successfully! ${JSON.stringify(result)}`)
        await producer.disconnect();
    }
    catch(ex)
    {
        console.error(`Something bad happened ${ex}`)
    }
    finally{
        process.exit(0);
    }


}

File: Lab 2/Owais Quadri/Python/producer.py
from sys import argv
from kafka import KafkaProducer
producer = KafkaProducer(bootstrap_servers=["localhost:9093","localhost:9094","localhost:9095"])
topic="Usernames"
message=argv[1]

if message[0] > 'N':
    producer.send(topic,bytearray(message.encode()),partition=1)
    print("Message sent to partition 1")
else:
    producer.send(topic,bytearray(message.encode()),partition=0)
    print("Message sent to partition 0")
